pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1577,7 +1577,6 @@ private void setPartitionParameters(RealtimeSegmentConfig.Builder realtimeSegmen
             _segmentLogger.warn(
                 "Number of stream partitions: {} does not match number of partitions in the partition config: {}, "
                     + "using number of stream " + "partitions", numPartitionGroups, numPartitions);
-            _serverMetrics.addMeteredTableValue(_tableNameWithType, ServerMeter.REALTIME_PARTITION_MISMATCH, 1);
```

Comment:
Emitting the metric here is wrong, because numPartitions fetched form the stream is used if there's a mismatch between table config and stream metadata. So in reality no event with incorrect partition is consumed.

Reply:
Also, we should change the logger to an info log.

Better yet, drop the comparison altogether between the tableconfig value and the number of partitions we come across in the stream.  Two reasons:
- The tableconfig value is not used while routing queries to realtime tables
- Stream partitions change dynamically, and we don't update the table config when we detect new partitions. So, it is impossible to keep them in sync.

Reply:
I believe the best action item here is to drop numPartition from table config altogether as it shouldn't be used anywhere, and does not provide any value.
To keep the scope of this PR limited to the fix, I'm just changing the log level to info.

Reply:
numPartitions in the table config seems to be used for instance assignment when a table is created (or rebalanced). I am not able to determine easily how the actual number of partitions in the topic is used for instance creation instead of the configured value (for realtime tables). The value is also used for table rebalancing. 

Lastly, realtime and offline tables have the same table config definition.

So we cannot really remove the num partitions from table config definitions.

But it is clear that there is no use for this field for realtime tables, and hence there is really no need to compare this with the actual number of partitions, and hence the suggestion to remove the compare.

@Jackie-Jiang what do you think?

I am also ok to keep it as an info, but I feel it is extraneous code. If we can remove it, we should.

Reply:
Taking a closer look at the logic and I feel we are making wrong assumptions based on Kafka behavior. `numPartitions` is used to let pinot know how the data is supposed to be partitioned. For Kafka, data is always partitioned by the upstream Kafka partition, but this assumption doesn't hold for all streams. The clean solution is to introduce an interface in the stream metadata provider to provide the actual `numPartitions` to use by passing in the value from the table config.

Reply:
@Jackie-Jiang are you ok with creating an issue to clean this up, because we are facing this problem in production now.